cl/cltypes/aggregate.go

```diff
@@ -54,6 +55,17 @@ func (a *AggregateAndProof) HashSSZ() ([32]byte, error) {
 	return merkle_tree.HashTreeRoot(a.AggregatorIndex, a.Aggregate, a.SelectionProof[:])
 }
 
+// SignedAggregateAndProofData is passed to SignedAggregateAndProof service. The service does the signature verification
+// asynchronously. That's why we cannot wait for its ProcessMessage call to finish to check error. The service
+// will do re-publishing of the gossip or banning the peer in case of invalid signature by itself.
+// that's why we are passing sentinel.SentinelClient and *sentinel.GossipData to enable the service
+// to do all of that by itself.
+type SignedAggregateAndProofData struct {
+	SignedAggregateAndProof *SignedAggregateAndProof
+	SentinelClient          sentinel.SentinelClient
```

Comment:
Idea, move sentinelClient to `AggregateAndProof` Service.
Additionally this struct can be avoided with `PeeredObject`

Reply:
I didn't touch `AggregateAndProof` struct because of this  line `obj.SignedAggregateAndProof.DecodeSSZ(data.Data, int(version));`, if I changed fields maybe the decoding does'n work. Also `PeeredObject` has peer as string, in the `GossipData` peer pointer is ready for the call `BanPeer`.

Reply:
removed `SentinelClient          sentinel.SentinelClient` from the struct